### PR TITLE
[sharedb] `sendPresence` middleware

### PR DESCRIPTION
**Pull request by alecgibson**
This change adds type support for the [`sendPresence`][1] middleware
hook, with a typed [`PresenceMessage`][2].

[1]: https://github.com/share/sharedb/blob/07f8bd98e5cdae37e925d4c0c709a1c3cbbca5e0/lib/agent.js#L816
[2]: https://github.com/share/sharedb/blob/07f8bd98e5cdae37e925d4c0c709a1c3cbbca5e0/lib/agent.js#L755-L767

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

### CODE REVIEW

File: types/sharedb/index.d.ts [hunk 1]
```diff
@@ -326,4 +332,17 @@ interface GetOpsOptions {
...
+    a: 'p';
+    ch: string; // channel
+    src: string; // client ID
+    id: string; // presence ID
+    p: any; // presence payload
+    pv: number; // presence version
+    c?: string; // document collection
+    d?: string; // document ID
+    v?: number; // document number
+    t?: string; // document type
```
Comment (ericyhwang) [inline types/sharedb/index.d.ts:345]: I'd suggest going with `// document version` and `// document OT type` to make their meanings clearer. Though we don't even have full comments on these in our JS source 😅

**Review by ericyhwang — APPROVED** [review]

File: types/sharedb/index.d.ts (hunk 1 above)
Comment (alecgibson) [reply types/sharedb/index.d.ts:345]: I actually meant to write `version`, just had a brain burp. Good catch.

**Review by ericyhwang — APPROVED** [review]
Re-approving after update to the comments on the types

**Comment by alecgibson** [comment]
Ready to merge

**Pull request merged**